Add docs for the Trending Metrics feature (#938)
grafana/k6-docs: src/data/markdown/docs/03 cloud/02 Analyzing Results/08 Test Comparison.md

![k6 Cloud Results: HTTP comparison](./images/07-Test-Comparison/http-comparison.png)

## Compare all runs for a test

To compare runs for a test across time, use the *performance-trending chart*.

> The chart displayes test run metrics and uses colors to signal the status of a specific run.

Reviewer: ```suggestion
The chart displays test-run metrics, using colors to signal the status of a specific run.
```

To view the performance-trending chart for multiple tests belonging to a same project, open the Project page:

![Project Performance Trending](./images/09-Performance-Trending/perf-trending-project.png)

Additionally, in order to view the performance-trending chart for an individual test, open the test's page:

![Test Performance Trending](./images/09-Performance-Trending/perf-trending-test.png)

This last version of the chart shows more data points over time. For more information on an individual run, hover over any bar in the chart.

> The data displayed in the performance-trending chart is `p95` of the HTTP response time (`http_req_time`) for responses with a status code (so, for example, requests that timed out are not considered).

Reviewer: Is the default?

> The data displayed in the performance-trending chart is `p95` of the HTTP response time (`http_req_time`) for responses with a status code (so, for example, requests that timed out are not considered).

Author: Yes! I will mention that

> We call this combination of metric plus aggregation function the _trending metric_ of the chart. k6 produces a single value for each test run using the trending metric, and then plots each value the chart.

Author: I want to maintain the term "trending metric" when possible, so I will write it like this:
summary _trending metric_
wdyt?

> We call this combination of metric plus aggregation function the _trending metric_ of the chart. k6 produces a single value for each test run using the trending metric, and then plots each value the chart.

Author: Also changing "test" for "test run"

> We call this combination of metric plus aggregation function the _trending metric_ of the chart. k6 produces a single value for each test run using the trending metric, and then plots each value the chart.

Reviewer: Whether to italicize the whole phrase or just _trend metric_ is not something I feel strongly about though haha.

> We call this combination of metric plus aggregation function the _trending metric_ of the chart. k6 produces a single value for each test run using the trending metric, and then plots each value the chart.

Author: I agree with your explanation, it's just that the UI uses the term "trending metric" on many different places so I thought it would be best to at least mention the term explicitly once in the docs, to make it feel more connected.

> We call this combination of metric plus aggregation function the _trending metric_ of the chart. k6 produces a single value for each test run using the trending metric, and then plots each value the chart.

Reviewer: Oh, I see I didn't think about the UI text. I think your proposal is reasonable here. I updated my suggestion.

It is possible to customize the trending metric on a per-test basis, which allows displaying different information in the test's performance-trending chart.

### Customizing the trending metric 

To customize the trending metric used for a test, navigate to the Project page, and then click on the three dots at the top-right corner of the test's performance-trending chart. Then, select the "Trending metric" option:

![Customizing Trending Metric Step 1](./images/09-Performance-Trending/trending-metric-tutorial-1.png)

> This will bring up the "Customize Trending Metric" window:

Reviewer: ```suggestion
This brings up a window:
```

![Customizing Trending Metric Step 2](./images/09-Performance-Trending/trending-metric-tutorial-2.png)

From here, you can select the metric to use in the performance-trending chart. You'll notice that both standard (created by all k6 test runs) and custom (user-defined) metrics are listed.

> For this example, we will select "CPU Usage", and then advance to the next step:

Reviewer: ```suggestion
This example selects `CPU Usage`:
```

![Customizing Trending Metric Step 3](./images/09-Performance-Trending/trending-metric-tutorial-3.png)

> In this section, you can select:

Reviewer: ```suggestion
After you select a metric, you can select:
```

- The aggregation function to apply to the metric (in this case, "Average" is selected).
- A set of tags & tag values used to filter the metric's values (in this case, selecting values only from instances in the `amazon:us:ashburn` load zone).

> Note that you can reset the configuration to use the default trending metric by clicking on `Reset to default trending metric`. This is only possible after you have made modifications to the default configuration.

Reviewer: ```suggestion
To reset the configuration, use the default trending metric with **Reset to default trending metric**.
```

Reviewer: I don't think it's necessary. To say when users can reset.

> Once the desired parameters have been selected, press "Save" to apply the changes. k6 will calculate the required values and then plot them in the performance-trending chart.

Reviewer: ```suggestion
After you select the desired parameters, **Save** to apply the changes. k6 calculates the required values, then plots them in the performance-trending chart.
```

> For the example steps describe previously, the result will look like this:

Reviewer: ```suggestion
In this example, the results look like this:
```

![k6 Cloud Results: HTTP comparison](./images/07-Test-Comparison/http-comparison.png)

> ## Compare all runs for a test

Reviewer: @federicotdn I'm adding this suggestion. I realized that you can also compare tests by failure in the top-level page of a project, so I wanted to make the heading more specific.
